diff -ruN memtest86+-5.01.orig/init.c memtest86+-5.01/init.c
--- memtest86+-5.01.orig/init.c	2013-08-09 19:26:27.000000000 -0700
+++ memtest86+-5.01/init.c	2014-01-29 22:25:30.681221015 -0800
@@ -164,13 +164,12 @@
 	for(i=0, pp=(char *)(SCREEN_ADR+1); i<TITLE_WIDTH; i++, pp+=2) {
 		*pp = 0x20;
 	}
-	cprint(0, 0, "      Memtest86  5.01        ");
-
+	cprint(0, 0, "  Memtest86  5.01 (Finnix)   ");
 	/* Set Blinking "+" */
-	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=30) {
+	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=22) {
 		*pp = 0xA4;
 	}
-	cprint(0, 15, "+");
+	cprint(0, 11, "+");
 
 	/* Do reverse video for the bottom display line */
 	for(i=0, pp=(char *)(SCREEN_ADR+1+(24 * 160)); i<80; i++, pp+=2) {
