diff -ruN memtest86+-4.20.orig/init.c memtest86+-4.20/init.c
--- memtest86+-4.20.orig/init.c	2011-01-23 10:11:04.000000000 -0800
+++ memtest86+-4.20/init.c	2013-03-10 15:42:59.145466408 -0700
@@ -112,12 +112,12 @@
 	for(i=0, pp=(char *)(SCREEN_ADR+1); i<TITLE_WIDTH; i++, pp+=2) {
 		*pp = 0x20;
 	}
-	cprint(0, 0, "      Memtest86  v4.20      ");
+	cprint(0, 0, " Memtest86  v4.20  (Finnix) ");
 
-	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=30) {
+	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=20) {
 		*pp = 0xA4;
 	}
-	cprint(0, 15, "+");
+	cprint(0, 10, "+");
 
 	/* Do reverse video for the bottom display line */
 	for(i=0, pp=(char *)(SCREEN_ADR+1+(24 * 160)); i<80; i++, pp+=2) {
